hadoop/src/main/java/org/apache/carbondata/hadoop/util/Hive2CarbonExpression.java

```diff
@@ -0,0 +1,183 @@
+import org.apache.hadoop.hive.ql.udf.generic.GenericUDFIn;
+import org.apache.hadoop.hive.ql.udf.generic.GenericUDFOPAnd;
+import org.apache.hadoop.hive.ql.udf.generic.GenericUDFOPEqual;
+import org.apache.hadoop.hive.ql.udf.generic.GenericUDFOPEqualOrGreaterThan;
+import org.apache.hadoop.hive.ql.udf.generic.GenericUDFOPEqualOrLessThan;
+import org.apache.hadoop.hive.ql.udf.generic.GenericUDFOPGreaterThan;
+import org.apache.hadoop.hive.ql.udf.generic.GenericUDFOPLessThan;
+import org.apache.hadoop.hive.ql.udf.generic.GenericUDFOPNotEqual;
+import org.apache.hadoop.hive.ql.udf.generic.GenericUDFOPNotNull;
+import org.apache.hadoop.hive.ql.udf.generic.GenericUDFOPNull;
+import org.apache.hadoop.hive.ql.udf.generic.GenericUDFOPOr;
+import org.apache.log4j.Logger;
+
+/**
+ * @description: hive expression to carbon expression
+ */
+public class Hive2CarbonExpression {
+  public static final int left = 0;
+  public static final int right = 1;
+  private static final Logger LOG =
+      LogServiceFactory.getLogService(CarbonInputFormat.class.getName());
+
+  public static Expression convertExprHive2Carbon(ExprNodeDesc exprNodeDesc) {
+    if (exprNodeDesc instanceof ExprNodeGenericFuncDesc) {
+      ExprNodeGenericFuncDesc exprNodeGenericFuncDesc = (ExprNodeGenericFuncDesc) exprNodeDesc;
+      GenericUDF udf = exprNodeGenericFuncDesc.getGenericUDF();
+      List<ExprNodeDesc> ll = exprNodeGenericFuncDesc.getChildren();
+      if (udf instanceof GenericUDFIn) {
+        ColumnExpression columnExpression = new ColumnExpression(ll.get(left).getCols().get(left),
+            getDateType(ll.get(left).getTypeString()));
+        List<Expression> listExpr = new ArrayList<>();
+        for (int i = right; i < ll.size(); i++) {
+          LiteralExpression literalExpression = new LiteralExpression(ll.get(i).getExprString(),
+              getDateType(ll.get(left).getTypeString()));
+          listExpr.add(literalExpression);
+        }
+        ListExpression listExpression = new ListExpression(listExpr);
+        return new InExpression(columnExpression, listExpression);
+      } else if (udf instanceof GenericUDFOPOr) {
+        Expression leftExpression =
+            convertExprHive2Carbon(exprNodeGenericFuncDesc.getChildren().get(left));
+        Expression rightExpression =
+            convertExprHive2Carbon(exprNodeGenericFuncDesc.getChildren().get(right));
+        return new OrExpression(leftExpression, rightExpression);
+      } else if (udf instanceof GenericUDFOPAnd) {
+        Expression leftExpression =
+            convertExprHive2Carbon(exprNodeGenericFuncDesc.getChildren().get(left));
+        Expression rightExpression =
+            convertExprHive2Carbon(exprNodeGenericFuncDesc.getChildren().get(right));
+        return new AndExpression(leftExpression, rightExpression);
+
+      } else if (udf instanceof GenericUDFOPEqual) {
+        ColumnExpression columnExpression = null;
+        if (ll.get(left) instanceof ExprNodeFieldDesc) {
```

Comment:
will this handle for all the complex date types like STRUCT, ARRAY and MAP?
Please add test cases where it tries to create filter expression for all the complex data types.